recipe/meta.yaml

```diff
@@ -32,7 +32,7 @@ requirements:
     - python
     - pip
     - setuptools
-    - llvmlite >=0.30.0
+    - llvmlite >=0.31.0
```

Comment:
@seibert, `numba::numba` has `llvmlite 0.31.*`. I don't know the backward compatibilty guarantees of `llvmlite`. Should we do the same here?

Reply:
(@henryiii, this not a blocker for this PR, just me being curious ;).)